src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -17,24 +17,24 @@ class CVulkanPhysicalDevice final : public IPhysicalDevice
     {
         // Get Supported Extensions
         {
-            uint32_t  count;
-            // Get Count First and Resize
+            uint32_t count;
             VkResult res = vkEnumerateDeviceExtensionProperties(m_vkPhysicalDevice, nullptr, &count, nullptr);
             assert(VK_SUCCESS == res);
-            supportedExtensions.resize(count); 
-            // Now fill the Vector
-            res = vkEnumerateDeviceExtensionProperties(m_vkPhysicalDevice, nullptr, &count, supportedExtensions.data());
+            core::vector<VkExtensionProperties> vk_extensions(count);
+            res = vkEnumerateDeviceExtensionProperties(m_vkPhysicalDevice, nullptr, &count, vk_extensions.data());
             assert(VK_SUCCESS == res);
```

Comment:
Seeing as this can fail, should I create a `static` create method for `CVulkanPhysicalDevice` as well @devshgraphicsprogramming ?

Reply:
no, because what the physical device supports is immutable

what can fail is creating a Logical Device with the extensions the user required

Reply:
Okay.
Resolved.